app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  include SessionsHelper
+  before_action :logged_in?, only: %i[new create]
+
+  def index
+    @posts = Post.all
+  end
+
+  def show
+    @post = Post.find(params[:format])
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
```

Comment:
It is better if you use the method generated by the association: `current_user.post.build(post_params)`.